lib/ddtrace/contrib/sinatra/request_span.rb

```diff
@@ -0,0 +1,39 @@
+module Datadog
+  module Contrib
+    module Sinatra
+      # Middleware used for automatically tagging configured headers and handle request span
+      module RequestSpan
+        SINATRA_REQUEST_SPAN = 'datadog.sinatra_request_span'.freeze
+        SINATRA_REQUEST_TRACE_NAME = 'sinatra.request'.freeze
+
+        module_function
+
+        def span!(env)
+          env[SINATRA_REQUEST_SPAN] ||= build_span(env)
+        end
+
+        def fetch_span(env)
```

Comment:
Is there a functional difference between this and `#span!`? Is there some place in our code you'd want to `fetch` a span and get nothing back? And that would be okay?

If not, couldn't you just use `#span!`?

Reply:
@pawelchcki thoughts here?

Reply:
I added `fetch_span` to support the existing case in `Tracer` response filter, where an error is logged if span doesn't exist. (Not sure how valid that approach is, but It wasn't something I wanted to change at this point).

`span!` will populate the `env` if the span does not exist hence `!` to indicate that. `fetch_span` is only meant to hide implementation detail of `env[SINATRA_REQUEST_SPAN]`